param/__init__.py

```diff
@@ -489,6 +490,7 @@ def __get__(self,obj,objtype):
             return self._produce_value(gen)
 
 
+    @instance_descriptor
```

Comment:
Looks like this `instance_descriptor` decorator is used precisely once. Where else would you expect it to be used?

Reply:
It should actually also be used on ``Parameter.__set__``, think that's an oversight.

Reply:
It does seem to be used there, must have missed it in the diff.